src/NetworkDevice.cpp

```diff
@@ -0,0 +1,19 @@
+#include "NetworkDevice.hpp"
+
+NetworkDevice::NetworkDevice(DeviceType type, double processing_delay_ns):
```

Comment:
Why do we need `processing_delay_ns`? 
1. Usually it is small and can be neglected compared to other types of delays.
2. When it is not small, this can just be added to the propagation delay to the next hop.